mirgecom/flux.py

```diff
@@ -30,9 +33,94 @@
 THE SOFTWARE.
 """
 import numpy as np  # noqa
+from meshmode.dof_array import DOFArray
+from mirgecom.fluid import (
+    ConservedVars,
+    make_conserved
+)
 
 
-def lfr_flux(cv_tpair, f_tpair, normal, lam):
+def gradient_flux_central(u_tpair, normal):
+    r"""Compute a central flux for the gradient operator.
+
+    The central gradient flux, $\mathbf{h}$, of a scalar quantity $u$ is calculated
+    as:
+
+    .. math::
+
+        \mathbf{h}({u}^-, {u}^+; \mathbf{n}) = \frac{1}{2}
+        \left({u}^{+}+{u}^{-}\right)\mathbf{\hat{n}}
+
+    where ${u}^-, {u}^+$, are the scalar function values on the interior
+    and exterior of the face on which the central flux is to be calculated, and
+    $\mathbf{\hat{n}}$ is the *normal* vector.
+
+    *u_tpair* is the :class:`~grudge.trace_pair.TracePair` representing the scalar
+    quantities ${u}^-, {u}^+$. *u_tpair* may also represent a vector-quantity
+    :class:`~grudge.trace_pair.TracePair`, and in this case the central scalar flux
+    is computed on each component of the vector quantity as an independent scalar.
+
+    Parameters
+    ----------
+    u_tpair: :class:`~grudge.trace_pair.TracePair`
+        Trace pair for the face upon which flux calculation is to be performed
+    normal: numpy.ndarray
+        object array of :class:`~meshmode.dof_array.DOFArray` with outward-pointing
+        normals
+
+    Returns
+    -------
+    numpy.ndarray
+        object array of :class:`~meshmode.dof_array.DOFArray` with the flux for each
+        scalar component.
+    """
+    tp_avg = u_tpair.avg
+    tp_join = tp_avg
+    if isinstance(tp_avg, DOFArray):
```

Comment:
Leave a `FIXME` here to make sure this gets improved once generalized `outer` arrives. cc @majosm